plugins/pixart-rf/fu-pxi-common.h

```diff
@@ -9,7 +9,39 @@
 
 #include <fwupdplugin.h>
 
+/* OTA spec check result */
+enum ota_spec_check_result {
+	OTA_SPEC_CHECK_OK		= 1,	/* Spec check ok */
+	OTA_FW_OUT_OF_BOUNDS		= 2,	/* OTA firmware size out of bound */
+	OTA_PROCESS_ILLEGAL		= 3,	/* Illegal OTA process */
+	OTA_RECONNECT			= 4,	/* Inform OTA app do reconnect */
+	OTA_FW_IMG_VERSION_ERROR	= 5,	/* FW image file version check error */
+	OTA_DEVICE_LOW_BATTERY		= 6,	/* Device is under low battery */
+	OTA_SPEC_CHECK_MAX_NUM,			/* Max number of OTA driver defined error code */
+};
+
+struct ota_state {
```

Comment:
@jyu99 or ota_fw_state perhaps?

Reply:
ota_fw_state  is better , thanks

Reply:
Could I expand the this plugin for the BLE device ? I mean use the fu_bluez_device rather than fu_udev_device.

Reply:
You could use the bluez ble support, but it would be a new object (FuPxiBluezDevice perhaps) -- but I'm not sure what extra it adds -- the BLE devices already work, no?

Reply:
Thanks your reply. I want to add the FuPxiBluezDevice  in future , but  I'm not sure the FuPxiBluezDevice  can add to pixart_rf plugin. Beacuse our plugin register the hidraw rather than bluetooth.

Reply:
we can discuss after add wireless peripheral and dongle receiver. thanks.